ui/components/nance/RetroactiveRewards.tsx

```diff
@@ -0,0 +1,394 @@
+          d.address.toLowerCase() === userAddress.toLowerCase()
+        ) {
+          setDistribution(d.distribution)
+          setEdit(true)
+          break
+        }
+      }
+    }
+  }, [userAddress, distributions])
+  const handleDistributionChange = (projectId: string, value: number) => {
+    setDistribution((prev) => ({
+      ...prev,
+      [projectId]: Math.min(100, Math.max(1, value)),
+    }))
+  }
+
+  const addresses = distributions ? distributions.map((d) => d.address) : []
+
+  const SNAPSHOT_ID =
+    '0xa38f7cfeb73b166aea0b65432230bc19faf5411e7f86cc8ea3b961d7c72c85ed'
+  const { data: _vps } = useVotingPowers(
+    addresses,
+    SNAPSHOT_SPACE_NAME,
+    SNAPSHOT_ID
+  )
+  const votingPowers = _vps ? _vps.map((vp) => (vp ? vp.vp : 0)) : []
+  const addressToQuadraticVotingPower = Object.fromEntries(
+    addresses.map((address, i) => [address, Math.sqrt(votingPowers[i])])
+  )
+  const votingPowerSumIsNonZero =
+    _.sum(Object.values(addressToQuadraticVotingPower)) > 0
+  const userHasVotingPower =
+    userAddress &&
+    userAddress in addressToQuadraticVotingPower &&
+    addressToQuadraticVotingPower[userAddress] > 0
+
+  // All projects need at least one citizen distribution to do iterative normalization
+  const isCitizens = useCitizens(chain, addresses)
+  const citizenDistributions = distributions?.filter((_, i) => isCitizens[i])
+  const nonCitizenDistributions = distributions?.filter(
+    (_, i) => !isCitizens[i]
+  )
+  const allProjectsHaveCitizenDistribution = projects.every(({ id }) =>
+    citizenDistributions.some(({ distribution }) => id in distribution)
+  )
+  const readyToRunVoting =
+    allProjectsHaveCitizenDistribution && votingPowerSumIsNonZero
+
+  const projectIdToEstimatedPercentage: { [key: string]: number } =
+    readyToRunVoting
+      ? computeRewardPercentages(
+          citizenDistributions,
+          nonCitizenDistributions,
+          projects,
+          addressToQuadraticVotingPower
+        )
+      : {}
+
+  const { contract: distributionTableContract } = useContract(
+    DISTRIBUTION_TABLE_ADDRESSES[chain.slug]
+  )
+  const { tokens } = useAssets()
+  const { ethBudget, usdBudget, mooneyBudget, ethPrice } = getBudget(
+    tokens,
+    year,
+    quarter
+  )
+  const {
+    projectIdToETHPayout,
+    projectIdToMooneyPayout,
+    addressToEthPayout,
+    addressToMooneyPayout,
+    ethPayoutCSV,
+    mooneyPayoutCSV,
+  } = getPayouts(
+    projectIdToEstimatedPercentage,
+    projects,
+    ethBudget,
+    mooneyBudget
+  )
+
+  const handleSubmit = async () => {
+    const totalPercentage = Object.values(distribution).reduce(
+      (sum, value) => sum + value,
+      0
+    )
+    if (totalPercentage !== 100) {
+      toast.error('Total distribution must equal 100%', {
+        style: toastStyle,
+      })
+      return
+    }
+    // Fill in empty values with 0
+    // Missing values are assumed to be projects which the user is a contributor to
```

Comment:
Just checking - if someone explicitly enters a "0" this will be handled differently than an empty value right?

Reply:
i think the desired behavior we want is, if you fill in a 0, that's a 0. if you leave it blank, we will fill in with iterative normalization. updated the code to reflect this.